spec/unit/controllers/services/service_keys_controller_spec.rb

```diff
@@ -0,0 +1,227 @@
+require 'spec_helper'
+
+module VCAP::CloudController
+  describe ServiceKeysController do
+    describe 'Attributes' do
+      it do
+        expect(described_class).to have_creatable_attributes({
+                                                                 name: { type: 'string', required: true },
+                                                                 service_instance_guid: { type: 'string', required: true }
+                                                             })
+      end
+    end
+
+    CREDENTIALS = { 'foo' => 'bar' }
+
+    let(:guid_pattern) { '[[:alnum:]-]+' }
+    let(:bind_status) { 200 }
+    let(:bind_body) { { credentials: CREDENTIALS } }
+    let(:unbind_status) { 200 }
+    let(:unbind_body) { {} }
+
+    def broker_url(broker)
+      base_broker_uri = URI.parse(broker.broker_url)
+      base_broker_uri.user = broker.auth_username
+      base_broker_uri.password = broker.auth_password
+      base_broker_uri.to_s
+    end
+
+    def stub_requests(broker)
+      stub_request(:put, %r{#{broker_url(broker)}/v2/service_instances/#{guid_pattern}/service_bindings/#{guid_pattern}}).
+          to_return(status: bind_status, body: bind_body.to_json)
+    end
+
+    def bind_url_regex(opts={})
+      service_binding = opts[:service_binding]
+      service_binding_guid = service_binding.try(:guid) || guid_pattern
+      service_instance = opts[:service_instance] || service_binding.try(:service_instance)
+      service_instance_guid = service_instance.try(:guid) || guid_pattern
+      broker = opts[:service_broker] || service_instance.service_plan.service.service_broker
+      %r{#{broker_url(broker)}/v2/service_instances/#{service_instance_guid}/service_bindings/#{service_binding_guid}}
+    end
+
+    describe 'create' do
+      context 'for managed instances' do
+        let(:instance) { ManagedServiceInstance.make }
+        let(:space) { instance.space }
+        let(:service) { instance.service }
+        let(:developer) { make_developer_for_space(space) }
+        let(:name) { 'fake-service-key' }
+        let(:service_instance_guid) { instance.guid }
+        let(:req) {
+          {
+              name: :name,
```

Comment:
I think instead of `:name`, you want `name`. This means that you will uses the name created by the `let(:name) ...` above.